### PR TITLE
RenderTimeline element

### PR DESCRIPTION
**Pull request by pmconne**
Introduce RenderTimeline element which hosts a schedule script.
Clean up and document RenderSchedule APIs. In particular, make interfaces reflect the fact that the `value` properties of timeline entries can be null or undefined.
Deprecate `DisplayStyleSettings.scheduleScript` which embeds a potentially massive script in display style's JSON properties, in favor of `DisplayStyleSettings.renderTimeline` which specifies the Id of the RenderTimeline element hosting the script.
Ensure cloning a display style keeps the schedule script intact, including the Id of the element (a display style or render timeline) hosting the script.
Delete the synchro-schedule-importer test app that was used to bootstrap support for schedule simulation.
Delete iot-demo that was only used for Microsoft demo and has since just been a maintenance burden.

[Feature](https://dev.azure.com/bentleycs/iModelTechnologies/_workitems/edit/552822).
[Bug](https://dev.azure.com/bentleycs/iModelTechnologies/_workitems/edit/552826).
[Corresponding native PR](https://dev.azure.com/bentleycs/iModelTechnologies/_git/imodel02/pullrequest/159609).

### CODE REVIEW

**Review by mgooding — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

**Review by scsewall — APPROVED** [review]

File: common/api/presentation-components.api.md [hunk 1]
```diff
@@ -177,10 +177,10 @@ export interface DataProvidersFactoryProps {
 export const DEFAULT_PROPERTY_GRID_RULESET: Ruleset;
 
 // @beta @deprecated
-export function DEPRECATED_controlledTreeWithFilteringSupport<P extends ControlledTreeWithVisibleNodesProps>(TreeComponent: React.FC<P>): React.FC<Pick<P & ControlledTreeWithFilteringSupportProps, "filter" | "onFilterApplied" | "onMatchesCounted" | "activeMatchIndex" | "nodeLoader" | "onNodeLoaderChanged" | Exclude<keyof P, "visibleNodes">>>;
+export function DEPRECATED_controlledTreeWithFilteringSupport<P extends ControlledTreeWithVisibleNodesProps>(TreeComponent: React.FC<P>): React.FC<Pick<P & ControlledTreeWithFilteringSupportProps, "filter" | "nodeLoader" | "onFilterApplied" | "onMatchesCounted" | "onNodeLoaderChanged" | "activeMatchIndex" | Exclude<keyof P, "visibleNodes">>>;
```
Comment (RBBentley) [inline common/api/presentation-components.api.md:181]: Extract-api flakiness -- probably should be reverted.

File: common/api/ui-framework.api.md [hunk 2]
```diff
@@ -1910,7 +1910,7 @@ export const expandWidget: (base: {
         readonly home: {
             readonly widgetIndex: number;
             readonly widgetId: string | undefined;
-            readonly side: PanelSide;
+            readonly side: "left" | "right" | "bottom" | "top";
```
Comment (RBBentley) [inline common/api/ui-framework.api.md:1913]: More extract-api noise...

**Review by RBBentley — APPROVED** [review]
Looks good. - Will old saved views embedded scripts embedded in display style continue to work?   Adding some additional views for ImageTests would be worthwhile to detect if the schedule images differ.

File: common/api/ui-framework.api.md (hunk 2 above)
Comment (pmconne) [reply common/api/ui-framework.api.md:1913]: Seems more likely the presentation folks forgot to run extract-api. Addon PR will catch any problems.

**Comment by pmconne** [comment]
> Will old saved views embedded scripts embedded in display style continue to work?

Yes.

**Pull request merged**